Test/Bool/Falsum.expected

```diff
@@ -5,3 +5,8 @@ query:
 (check-sat)
 
 result: unsat
+failed to reconstruct proof: tactic 'apply' failed, failed to unify
+  (¬False ↔ True) → False
+with
```

Comment:
Is this a bug in proof reconstruction?

Reply:
The proofs generated by cvc5 sometimes contain hypotheses that are not in the original SMT query because of how theory rewrites are implemented internally. Those rewrites are currently added as hypotheses to `th0` after the goal (i.e., `th0 : Not goal -> rrh -> False`), which causes issues with unification. We are changing cvc5 printer to move `rrh` to the front (the whole interface between cvc5 and Lean is still a work-in-progress). Some of those hypotheses are simple enough to "simp" our way through them, but the reset need to be delegated to the user.